R/mean-model-workflow.R

```diff
@@ -100,14 +102,23 @@ MeanModelWorkflow <- R6::R6Class(
       )
       appendices <- appendices[file.exists(appendices)]
       if (length(appendices) > 0) {
-        mergeMarkdownFiles(appendices, self$reportFileName, logFolder = self$workflowFolder)
+        initialReportPath <- file.path(self$workflowFolder, self$reportFileName)
+        mergeMarkdownFiles(appendices, initialReportPath, logFolder = self$workflowFolder)
         renderReport(
-          self$reportFileName,
+          file.path(self$workflowFolder, self$reportFileName),
           logFolder = self$workflowFolder,
           createWordReport = self$createWordReport, 
           numberSections = self$numberSections,
           wordConversionTemplate = self$wordConversionTemplate
           )
+        # Move report if a non-default path is provided
+        copyReport(from = initialReportPath, to = self$reportFilePath, keep = TRUE)
+        if(self$createWordReport){
+          file.copy(
+            from = gsub(pattern = ".md", replacement = ".docx", x = initialReportPath), 
+            to = gsub(pattern = ".md", replacement = ".docx", x = self$reportFilePath)
+            )
+        }
```

Comment:
this code is exactly the same for all 3 workflow types. 
It would be better if e.g. the function `copyReport `would have additional argument (`copyWordReport `or so) and lines 117-120 would be moved into the copyReport function.
Then you would replace lines 115-121 with
`copyReport(from = initialReportPath, to = self$reportFilePath, keep = TRUE, copyWordReport =self$createWordReport)`
same for meanmodel and pop-workflow